feature-libs/order/components/order-details/order-detail-reorder/reorder-dialog/reorder-dialog.component.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import {
+  CartModification,
+  CartModificationList,
+  CartValidationStatusCode,
+  MultiCartFacade,
+} from '@spartacus/cart/base/root';
+import { OCC_CART_ID_CURRENT } from '@spartacus/core';
+import { ReorderOrderFacade } from '@spartacus/order/root';
+import {
+  FocusConfig,
+  ICON_TYPE,
+  LaunchDialogService,
+} from '@spartacus/storefront';
+import { BehaviorSubject } from 'rxjs';
+
+@Component({
+  selector: 'cx-reorder-dialog',
+  templateUrl: './reorder-dialog.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ReorderDialogComponent {
+  iconTypes = ICON_TYPE;
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: true,
+    focusOnEscape: true,
+  };
+
+  cartModifications: CartModification[] | undefined;
+
+  loading$ = new BehaviorSubject(false);
+  showDecisionPrompt$ = new BehaviorSubject(true);
```

Comment:
Why are these two attributes `BehaviorSubject`s.  To make things simpler, could they have been regular boolean attributes?

Reply:
It seems plain boolean vars don't work.  Perhaps an effect of the on push change detection strategy.